projects/npm-tools/packages/npm-scripts/src/sass/build.js

```diff
@@ -0,0 +1,181 @@
+/**
+ * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const fs = require('fs');
+const path = require('path');
+const sass = require('sass');
+
+const expandGlobs = require('../utils/expandGlobs');
+const log = require('../utils/log');
+
+const SASS_EXCLUDE = [
+	'**/build/**',
+	'**/classes/**',
+
+	// Copy from CSSBuilderArgs.java
+
+	'**/_diffs/**',
+	'**/.sass-cache*/**',
+	'**/.sass_cache_*/**',
+	'**/_sass_cache_*/**',
+	'**/_styled/**',
+	'**/_unstyled/**',
+	'**/css/aui/**',
+	'**/css/clay/**',
+	'**/tmp/**',
+];
+
+const CWD = process.cwd();
+
+const CSS_IMPORT_REGEX = /@import\s+url\s*\(\s*['"]?(.+\.css)/g;
```

Comment:
Probably more precise to do a non-greedy match here. Your regex would match:

```
@import url("a.css"); @import url("b.css");
^--------------------------------------^ match
             ^-------------------------^ capture group
```

ie. change it to:

```
/@import\s+url\s*\(\s*['"]?(.+?\.css)/g
                              ^
                              |
                      note the "?" here
```

Reply:
I copied this over from `CSSBuilderUtil.java`

```java
	private static final Pattern _cssImportPattern = Pattern.compile(
		"@import\\s+url\\s*\\(\\s*['\"]?(.+\\.css)");
```

Assuming my java->js conversion was accurate, do you think its worth changing this?

Reply:
Yes, that's why I said it.

The Java pattern obviously works, but it is fragile, because it makes assumptions about the input being well formed. By adding a single `?` character, you can make the regex more robust and handle badly formed input as well, so why not do it?

In an ideal world, we'd use AST-based transforms for everything and we would never have to worry about any of this. But in the actual world were we still use regexes for a lot of things, it's just a matter of best practice to tighten them up — in so far as is practical — to match what you want them to match and _not_ what you don't want them to match.

Reply:
Sounds good to me, just wanted to clarify and make sure we were good with making those sort of changes while "porting" over to new tech. I'll be sure to include it in a separate commit so we can as least have a reference to a change in the regex in case there are issues in the future.